hedera-node/hedera-app/src/main/java/com/hedera/node/app/blocks/impl/BlockStreamBuilder.java

```diff
@@ -0,0 +1,1055 @@
+    @NonNull
+    public BlockStreamBuilder ethereumHash(@NonNull final Bytes ethereumHash) {
+        this.ethereumHash = requireNonNull(ethereumHash, "ethereumHash must not be null");
+        ensureOutputBuilder();
+        return this;
+    }
+
+    @NonNull
+    @Override
+    public EthereumTransactionStreamBuilder feeChargedToPayer(long amount) {
+        transactionResultBuilder.transactionFeeCharged(transactionFee + amount);
+        return this;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    @NonNull
+    public BlockStreamBuilder paidStakingRewards(@NonNull final List<AccountAmount> paidStakingRewards) {
+        // These need not be externalized to block streams
+        this.paidStakingRewards = requireNonNull(paidStakingRewards, "paidStakingRewards must not be null");
+        transactionResultBuilder.paidStakingRewards(paidStakingRewards);
+        return this;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    @NonNull
+    public BlockStreamBuilder entropyNumber(final int num) {
+        ensureOutputBuilder()
+                .utilPrng(UtilPrngOutput.newBuilder().prngNumber(num).build());
+        return this;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    @NonNull
+    public BlockStreamBuilder entropyBytes(@NonNull final Bytes prngBytes) {
+        requireNonNull(prngBytes, "prngBytes must not be null");
+        ensureOutputBuilder()
+                .utilPrng(UtilPrngOutput.newBuilder().prngBytes(prngBytes).build());
+        return this;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    @NonNull
+    public BlockStreamBuilder evmAddress(@NonNull final Bytes evmAddress) {
+        // No-op
+        return this;
+    }
+
+    @Override
+    @NonNull
+    public List<AssessedCustomFee> getAssessedCustomFees() {
+        return assessedCustomFees;
+    }
+
+    // ------------------------------------------------------------------------------------------------------------------------
+    // fields needed for TransactionReceipt
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    @NonNull
+    public BlockStreamBuilder status(@NonNull final ResponseCodeEnum status) {
+        this.status = requireNonNull(status, "status must not be null");
+        transactionResultBuilder.status(status);
+        return this;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    @NonNull
+    public ResponseCodeEnum status() {
+        return status;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean hasContractResult() {
+        return this.contractFunctionResult != null;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public long getGasUsedForContractTxn() {
+        return this.contractFunctionResult.gasUsed();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    @NonNull
+    public BlockStreamBuilder accountID(@NonNull final AccountID accountID) {
+        // No-op
```

Comment:
Why are there a few no-ops in this file? E.g. this, `memo`, `fileID`...?

Reply:
These fields are externalized in `StateChanges` in the block stream, so in an item not created by this class.